results/cmd/api/main.go

```diff
@@ -161,3 +172,79 @@ func (s *server) DeleteTaskRunResult(ctx context.Context, req *pb.DeleteTaskRunR
 	}
 	return nil, nil
 }
+
+// ListTaskRunsResult receives a ListTaskRunRequest from users and return to users a list of TaskRuns according to query
+func (s *server) ListTaskRunsResult(ctx context.Context, req *pb.ListTaskRunsRequest) (*pb.ListTaskRunsResponse, error) {
+	env, err := cel.NewEnv(
+		cel.Types(&pb.TaskRun{}),
+		cel.Declarations(decls.NewIdent("taskrun", decls.NewObjectType("tekton.TaskRun"), nil)),
+	)
+	ast, issues := env.Compile(req.GetFilter())
+	if issues != nil && issues.Err() != nil {
+		log.Printf("type-check error: %s", issues.Err())
+		return nil, fmt.Errorf("failed to list taskruns: %w", err)
```

Comment:
We should return a more descriptive error. e.g. was it cause by the user? what should they do to resolve it?

We should also make sure to set a gRPC status code in the response.

Reply:
We should make similar changes for the errors below.

Reply:
done